bounce/server/resource/__init__.py

```diff
@@ -95,12 +95,20 @@ def set_defaults(schema, info, parent_key=None, parent_schema=None):
             if value == 'array':
                 sub_schema = schema['items']['properties']
                 # get the items in the array
-                items = info[parent_key]
-                for item in items:
-                    item = set_defaults(sub_schema, item, parent_key,
-                                        sub_schema)
-                # add updated items to info
-                info[parent_key] = items
+                if parent_key:
+                    items = info[parent_key]
+                    for item in items:
+                        item = set_defaults(sub_schema, item, parent_key,
+                                            sub_schema)
+                    # add updated items to info
+                    info[parent_key] = items
+                else:
+                    items = info
+                    for item in items:
+                        item = set_defaults(sub_schema, item, parent_key,
```

Comment:
It looks like you do this `for` loop in both cases here. Is there a way you can refactor this so you don't duplicate code?